### ISSUE TITLE
KeyError raised after calling remove_overlap

### ISSUE THREAD

**Issue by alexandersoto** [chaimleib/intervaltree#72]
Calling `remove_overlap` multiple times on the same tree can cause a KeyError. Here's the stack trace:
```
KeyError: Interval(1535742000.0, 1535743800.0)
  File "intervaltree/node.py", line 245, in remove_interval_helper
    self.s_center.remove(interval)

KeyError: Interval(1535742000.0, 1535743800.0)

  ...app specific stack trace

  File "intervaltree/intervaltree.py", line 473, in remove_overlap
    self.remove(iv)
  File "intervaltree/intervaltree.py", line 361, in remove
    self.top_node = self.top_node.remove(interval)
  File "intervaltree/node.py", line 211, in remove
    return self.remove_interval_helper(interval, done, should_raise_error=True)
  File "intervaltree/node.py", line 270, in remove_interval_helper
    self[direction] = self[direction].remove_interval_helper(interval, done, should_raise_error)
  File "intervaltree/node.py", line 248, in remove_interval_helper
    raise KeyError(interval)
```

Unfortunately, I don't have a failing test case. I'm using this library in a web app and haven't been able to recreate an offending tree/overlap that crashes. I believe it has something to do with when `remove_overlap` is called multiple times on the same tree, with a specific set of overlaps.

**Comment by iwismer** [comment]
I have the same issue, with the following tree:
```IntervalTree([Interval(0.0, 2.588, 841), Interval(65.5, 85.8, 844), Interval(93.6, 130.0, 837), Interval(125.0, 196.5, 829), Interval(391.8, 521.0, 825), Interval(720.0, 726.0, 834), Interval(800.0, 1033.0, 850), Interval(800.0, 1033.0, 855)])```
With the function call:
```tree.remove_overlap(0.0 521.0)```

It raises the error on the first call to the function.

**Comment by petersbr** [comment]
I'm also experiencing this issue, but during a remove_envelop call

**Comment by chaimleib** [comment]
Looks like this got fixed in 3.0.0.
